code/computational_geometry/jarvis_march/jarvis_march.py

```diff
@@ -0,0 +1,93 @@
+import matplotlib.pyplot as plt
+import numpy as np
+import math 
+
+class Point:
+    def __init__(self, x, y):
+        self.x = x
+        self.y = y
+
+    def equal(self, pt):
+        if self.x == pt.x and self.y == pt.y:
+            return True
+        return False
+
+    def __str__(self):
+        return "(" + str(self.x) + ","+str(self.y) + ")"
+
+
+def leftMostPoint(pts):
+    if not isinstance(pts, list):
+        return False
+    
+    curr = pts[0]
+    for pt in pts[1:]:
+        if pt.x == curr.x and pt.y < curr.y:
+            curr = pt
+        elif pt.x < curr.x and pt.x != curr.x:
+            curr = pt
+            
+    return curr
+
+def cross_product(a, b, c):
+    x1 = a.x - b.x
+    x2 = a.x - c.x
+    y1 = a.y - b.y
+    y2 = a.y - c.y
+    return y2 * x1 - y1 * x2
+
+def distance(p, p1, p2):
+    dist1 = math.hypot(p.x - p1.x, p.y - p1.y)
```

Comment:
You don't really need these variables, but this is just a recommendation.

Reply:
OK, fixed this.